src/ApplicationSampler.cpp

```diff
@@ -236,27 +294,28 @@ namespace geopm
         return m_short_region_buffer[event_signal];
     }
 
-    std::map<uint64_t, std::string> ApplicationSamplerImp::get_name_map(uint64_t name_key) const
-    {
-        throw Exception("ApplicationSamplerImp::" + std::string(__func__) + "() is not yet implemented",
-                        GEOPM_ERROR_NOT_IMPLEMENTED, __FILE__, __LINE__);
-        return {};
-    }
-
     uint64_t ApplicationSamplerImp::cpu_region_hash(int cpu_idx) const
     {
+        uint64_t result = GEOPM_REGION_HASH_INVALID;
         if (!m_status) {
-            return GEOPM_REGION_HASH_UNMARKED;
+            result = GEOPM_REGION_HASH_APP;
+        }
+        else if (m_is_cpu_active[cpu_idx]) {
+            result = m_status->get_hash(cpu_idx);
         }
-        return m_status->get_hash(cpu_idx);
+        return result;
     }
 
     uint64_t ApplicationSamplerImp::cpu_hint(int cpu_idx) const
     {
+        uint64_t result = GEOPM_REGION_HINT_INACTIVE;
         if (!m_status) {
-            return GEOPM_REGION_HINT_INACTIVE;
+            result = GEOPM_REGION_HINT_INACTIVE;
+        }
```

Comment:
Could simply delete this `if` statement, if the intention is to return `GEOPM_REGION_HINT_INACTIVE` two cases:
- `!m_status`
- `m_status && !m_is_cpu_active`